File: src/main/java/co/gatedaccess/web/repo/OccupantRepo.java
package co.gatedaccess.web.repo;

import co.gatedaccess.web.model.Occupant;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface OccupantRepo extends MongoRepository<Occupant, String> {
    Occupant findByGuardianCommunity_IdAndFirstNameOrLastNameContainingIgnoreCase(String id, String s1, String s2);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/OccupantRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.Occupant
import org.springframework.data.mongodb.repository.MongoRepository

interface OccupantRepo : MongoRepository<Occupant?, String?> {
    fun findByGuardianCommunityIdAndFirstNameOrLastNameContainingIgnoreCase(
        id: String?,
        s1: String?,
        s2: String?
    ): Occupant?
}
